doc/source/development/rfc/rfc85_policy_code_additions.rst

```diff
@@ -0,0 +1,88 @@
+.. _rfc-85:
+
+===========================================================
+RFC 85: Policy regarding substantial code additions
+===========================================================
+
+======== ==================================================
+Authors: Howard Butler, Even Rouault
+======== ==================================================
+Started: 2022-01-17
+Status:  Development
+======== ==================================================
+
+Summary
+-------
+
+This document describes the policies that the GDAL project will apply to assess
+substantial code additions, typically new drivers, in particular coming for new
+contributors to the project.
+
+Motivation
+----------
+
+The GDAL project has historically been quite open to new code additions, including
+drivers that rely on proprietary licensed and/or closed-source SDKs (called
+"binary SDK" in the rest of this document). This approach is part
+of its strengths, but also comes at a price, with the software being larger
+and larger, with sometimes contributions being abandoned by their authors without
+new maintainers.
+
+Being a free & open-source project, the GDAL project will apply stronger scrutiny to
+code that require a binary SDK and encourage contributors to submit code additions
+that do not depend on such binary SDK or license such SDK under free & open-source
+license terms.
+
+Policy
+------
+
+- Drivers require a designated responsible contact, tracked in
+  https://github.com/OSGeo/gdal/wiki/Maintainers-per-sub-system
+
+- Contributions should follow other RFCs describing development rules, and come
+  with test scripts and sufficient documentation, covering usage and build instructions.
+
+- If the driver require a binary SDK not downloadable without cost, or that requires
+  a complicated registration process, the GDAL team is unlikely to support
+  driver inclusion.
```

Comment:
From @gdt on gdal-dev:

```suggestion
- If the driver require a binary SDK not downloadable without cost, or that requires
  a complicated registration process, the GDAL team is unlikely to support
  driver inclusion. In addition, drivers depending on binary SDKs will default to off
  in the build system so that users who build GDAL will not accidentally create
  proprietary software. 
```

Though I think at the moment we do actually automatically build some drivers if we auto-detect their presence?

Reply:
Perhaps related, do we want something like this? 

```diff
+ - Any GDAL committer who requests it is expected to be provided with free access to binary SDKs
+   for the purposes of GDAL development.
```

(Bearing in mind that if I create an OSS driver for say, Kakadu or ODA as a licensee, I don't necessarily have the ability to fulfil this part. But it should be a consideration for the PSC)

Reply:
I've just responsed to Greg by email : "Sorry , but no. If people have put proprietary SDKs in default findable locations (which generally requires efforts since they generally have non standard layouts), they will be used by default, at least in the new CMake build. That makes things much more consistent. People can still disable them (see https://gdal.org/build_hints.html#cmake-package-dependent-options and https://gdal.org/build_hints.html#selection-of-drivers). People who build GDAL should look at the licensing of all dependencies. It might be well possible to create a GDAL build with only free components with incompatible licenses, who knows. Or maybe people want a build that is globally permissive licensed, and thus they must be careful of not including copyleft components. There was a RFC attempt at addressing this, but this was abandoned as being a too complex issue. "
Anyway build system considerations are out of the scope of this RFC

Reply:
> ```diff
> ```diff
> + - Any GDAL committer who requests it is expected to be provided with free access to binary SDKs
> +   for the purposes of GDAL development.
> ```

> (Bearing in mind that if I create an OSS driver for say, Kakadu or ODA as a licensee, I don't necessarily have the ability to fulfil this part. But it should be a consideration for the PSC)
> ```

for the reason you indicate, it is difficult to make it a requirement. It is clear that drivers that depend on SDKs that aren't freely available are quite prone to rot if they don't have a dedicated maintainer.